src/main/java/uk/gov/justice/hmpps/prison/service/SentenceEnvelopeService.kt

```diff
@@ -47,26 +41,26 @@ class SentenceEnvelopeService(
         pageRequest,
       )
     val activeBookings =
-      offenderBookingRepository.findAllByBookingIdIn(bookingIds.stream().map(OffenderBookingId::bookingId).toList())
-    val calculableSentenceEnvelopes = activeBookings.stream()
-      .map { offenderBooking: OffenderBooking -> determineCalculableSentenceEnvelope(offenderBooking) }
-      .toList()
+      offenderBookingRepository.findAllByBookingIdIn(bookingIds.map(OffenderBookingId::bookingId).toList())
+    val calculableSentenceEnvelopes =
+      activeBookings.map { determineCalculableSentenceEnvelope(it) }.sortedBy { it.bookingId }
     return PageImpl(calculableSentenceEnvelopes, pageRequest, bookingIds.totalElements)
   }
 
-  fun getCalculableSentenceEnvelopeByOffenderNumbers(offenderNumbers: Set<String?>?): List<CalculableSentenceEnvelope> {
+  fun getCalculableSentenceEnvelopeByOffenderNumbers(offenderNumbers: Set<String?>): List<CalculableSentenceEnvelope> {
     val bookingIds =
       offenderBookingRepository.findDistinctByActiveTrueAndOffenderNomsIdInAndSentences_statusAndSentences_CalculationType_CalculationTypeNotLikeAndSentences_CalculationType_CategoryNot(
         offenderNumbers,
         "A",
         "%AGG%",
         "LICENCE",
       )
-    val activeBookings =
-      offenderBookingRepository.findAllByBookingIdIn(bookingIds.stream().map(OffenderBookingId::bookingId).toList())
-    return activeBookings.stream()
-      .map { offenderBooking: OffenderBooking -> determineCalculableSentenceEnvelope(offenderBooking) }
-      .toList()
+
+    // ensure that the user has access to each of the bookings
+    bookingIds.forEach { bookingService.verifyBookingAccess(it.bookingId, false, "VIEW_PRISONER_DATA") }
```

Comment:
this was happening as a side effect of retrieving the booking again - moved out.  Would be nicer to have a `verifyBookingAccess` that actually takes a list instead...